pkgs/applications/window-managers/bspwm/git.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, libxcb, libXinerama, sxhkd, xcbutil, xcbutilkeysyms, xcbutilwm }:
+
+stdenv.mkDerivation rec {
+  name = "bspwm-git-2016-09-30";
+
+
+  src = fetchFromGitHub {
+    owner   = "baskerville";
+    repo    = "bspwm";
+    rev     = "8664c007e44de162c1597fd7e163635b274fb747";
+    sha256  = "0clvpz32z38i8kr10hqlifa661szpfn93c63m2aq2h4dwmr44slz";
+  };
+
+  buildInputs = [ libxcb libXinerama xcbutil xcbutilkeysyms xcbutilwm ];
+
+  buildPhase = ''
+    make PREFIX=$out
+  '';
+
+  installPhase = ''
+    make PREFIX=$out install
+  '';
+
+  meta = {
+    description = "A tiling window manager based on binary space partitioning (git version)";
+    homepage = http://github.com/baskerville/bspwm;
```

Comment:
Hey @cedeel, if I understand that correctly you're looking for some sort of documentation how `inherit` works (if not [my answer on your other PR](https://github.com/NixOS/nixpkgs/pull/19350#issuecomment-252435967) should already cover that).

You can find a chapter about the [Nix Expression Language](http://nixos.org/nix/manual/#ch-expression-language) in the [Nix manual](http://nixos.org/nix/manual/) which includes a section about [Inheriting attributes](http://nixos.org/nix/manual/#idm140737318140176) :wink:
There's also [A tour of Nix](https://nixcloud.io/tour/) which helped me a lot (I'm still new to Nix as well :smile: ).